File: temperaturas.cpp
#include <iostream>
using namespace std;


class temperatura{
	private:
		
		float celsiusFar(){
			float temp;
			cout<<"Quantos graus você quer converter? "<<endl;
			cin>>temp;
			temp = (temp * 1.80);
			return temp + 32;
			};
		
		float celsiusKelvin(){
			float temp;
			cout<<"Quantos graus você quer converter? "<<endl;
			cin>>temp;
			return temp + 273.15;
			};

		float farCelsius(){
			float temp;
                        cout<<"Quantos graus você quer converter? "<<endl;
			cin>>temp;
			temp = temp - 32 ;
			temp = temp * 5 ;
			return temp / 9;	
		};
		float farKelvin(){
                        float temp;
                        cout<<"Quantos graus você quer converter? "<<endl;
                        cin>>temp;
			temp -= 32;
			temp = temp * (5.0/9.0);
			return temp + 273.15;			
			};
		float kelvinCelsius(){
                        float temp;
                        cout<<"Quantos graus você quer converter? "<<endl;
                        cin>>temp;
			return temp - 273.15;		
			};
		float kelvinFar(){
                        float temp;
                        cout<<"Quantos graus você quer converter? "<<endl;
                        cin>>temp;
			temp = temp - 273.15;
			temp = temp * 1.80;
			return temp + 32;			
			};










	public:
		int caso;
		
		void conversor(){
				int convertida;

			std::cout<<"Qual temperatura você quer converter? "<<endl;
			std::cout<<"1- Celsius\n"<<
				   "2- Fahrenheit\n"<<
				   "3- Kelvin"<<endl;
			cin>>caso;
			
			switch (caso){
				case 1: //Celsius
					std::cout<<"Para qual você quer converter? "<<endl;
					std::cout<<"1- Fahrenheit\n"<<
					           "2- Kelvin"<<endl;
					cin>>convertida;
					if (convertida == 1){ 
						float a = celsiusFar();
						cout<<"A temperatura em fahrenheit: "<< a <<endl;
					}
					else{
						float a = celsiusKelvin();
						cout<<"A temperatura em Kelvin: "<< a <<endl;
					};
				case 2: //Fahrenheit
					//int convertida;
					std::cout<<"Para qual unidade você deseja converter? "<<endl;
					std::cout<<"1- Celsius\n"
						 <<"2- Kelvin"<<endl;
					cin>>convertida;
					if (convertida == 1){
						float a = farCelsius();
						cout<<"A temperatura em celsius é: "<<a<<endl;
					}
					else{
						float a = farKelvin();
						cout<<"A temperatura em Kelvin é: "<<a<<endl;
					};
				case 3: //Kelvin
					//int convertida;
					std::cout<<"Para qual unidade você quer converter?"<<endl;
					std::cout<<"1- Celsius\n"
						     <<"2- Fahrenheit"<<endl;
					cin>>convertida;
					if (convertida == 1){
						float a = kelvinCelsius();
						cout<<"A temperatura em celsius é "<< a <<endl;
					}
					else{
						float a = kelvinFar();
						cout<<"A temperatura em fahrenheit é "<<a<<endl;
					};

		
			};	
	};	
};

int main(){

	temperatura x;
	x.conversor();
	return 0;


};

File: conversores/Temperaturas.java
package conversores;
import java.util.Scanner;

public class Temperaturas {
    private Scanner scanner = new Scanner(System.in);

    private float celsiusToFahrenheit(float temp) {
        return (temp * 1.8f) + 32;
    }

    private float celsiusToKelvin(float temp) {
        return temp + 273.15f;
    }

    private float fahrenheitToCelsius(float temp) {
        return (temp - 32) * 5 / 9;
    }

    private float fahrenheitToKelvin(float temp) {
        return (temp - 32) * 5 / 9 + 273.15f;
    }

    private float kelvinToCelsius(float temp) {
        return temp - 273.15f;
    }

    private float kelvinToFahrenheit(float temp) {
        return (temp - 273.15f) * 1.8f + 32;
    }

    public void conversor() {
        System.out.println("Qual temperatura você quer converter? ");
        System.out.println("1- Celsius\n2- Fahrenheit\n3- Kelvin");
        int caso = scanner.nextInt();

        System.out.println("Quantos graus você quer converter? ");
        float temp = scanner.nextFloat();

        switch (caso) {
            case 1: // Celsius
                System.out.println("Para qual você quer converter? \n1- Fahrenheit\n2- Kelvin");
                int convertidaC = scanner.nextInt();
                if (convertidaC == 1) {
                    System.out.println("A temperatura em Fahrenheit: " + celsiusToFahrenheit(temp));
                } else {
                    System.out.println("A temperatura em Kelvin: " + celsiusToKelvin(temp));
                }
                break;

            case 2: // Fahrenheit
                System.out.println("Para qual unidade você deseja converter? \n1- Celsius\n2- Kelvin");
                int convertidaF = scanner.nextInt();
                if (convertidaF == 1) {
                    System.out.println("A temperatura em Celsius: " + fahrenheitToCelsius(temp));
                } else {
                    System.out.println("A temperatura em Kelvin: " + fahrenheitToKelvin(temp));
                }
                break;

            case 3: // Kelvin
                System.out.println("Para qual unidade você quer converter? \n1- Celsius\n2- Fahrenheit");
                int convertidaK = scanner.nextInt();
                if (convertidaK == 1) {
                    System.out.println("A temperatura em Celsius: " + kelvinToCelsius(temp));
                } else {
                    System.out.println("A temperatura em Fahrenheit: " + kelvinToFahrenheit(temp));
                }
                break;

            default:
                System.out.println("Opção inválida.");
        }
    }
}